### PR TITLE
Code cleaning

### PR DESCRIPTION
**Pull request by jjsjann123**
more code cleaning.
Still WIP and awaiting clang build.

### CODE REVIEW

File: third_party/nvfuser/test/test_gpu_fused_reduction.cpp [hunk 1]
```diff
@@ -2297,8 +2297,9 @@ TEST_F(NVFuserTest, FusionCrossIterationGroupedGridAllreduceWelfordShmoo_CUDA) {
     // Schedule the fusion as it will be done by the persistent
     // scheduler
 
-    auto input_cache = tv1;
-    auto output_cache = tv9->cacheBefore();
+    // auto input_cache = tv1;
```
Comment (naoyam) [inline third_party/nvfuser/test/test_gpu_fused_reduction.cpp]: nit: delete?

File: third_party/nvfuser/test/test_gpu_fused_reduction.cpp [hunk 2]
```diff
@@ -2287,7 +2286,8 @@ TEST_F(NVFuserTest, FusionCrossIterationGroupedGridAllreduceWelfordShmoo_CUDA) {
     auto tv4 = tvs.n;
     auto tv5 = broadcast(tv2, bcast_pattern);
     auto tv6 = broadcast(tv3, bcast_pattern);
-    auto tv7 = broadcast(tv4, bcast_pattern);
+    // auto tv7 = broadcast(tv4, bcast_pattern);
+    broadcast(tv4, bcast_pattern);
```
Comment (naoyam) [inline third_party/nvfuser/test/test_gpu_fused_reduction.cpp]: don't remember why `tv7` is not used, but I think this `broadcast` can be just removed.

**Review by naoyam — APPROVED** [review]
Thanks so much for this cleanup!

**Comment by jjsjann123** [comment]
I'm seeing segfault in my local tests... Something seems off... I don't think I changed any functionality... I'll try devel branch.

**Comment by jjsjann123** [comment]
failing tests doesn't look related. I'm merging this one.

**Pull request merged**